10_temperature_object/temperature.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class Temperature
+  def initialize(hash = {})
+    @hash = hash
+  end
+
+  def self.from_celsius(temp)
+    Temperature.new({ c: temp })
+  end
+
+  def self.from_fahrenheit(temp)
+    Temperature.new({ f: temp })
+  end
+
+  def in_fahrenheit
+    if @hash.key?(:f)
+      @hash[:f]
+    elsif @hash.key?(:c)
+      @hash[:c] * 9.to_f / 5 + 32
+    elsif @temp.is_a? Numeric
+      @temp * 9.to_f / 5 + 32
+    else
+      @temp
+    end
+  end
+
+  def in_celsius
+    if @hash.key?(:c)
+      '???????????????????'
+    elsif @hash.key?(:f)
+      (@hash[:f] - 32) * 5.to_f / 9
+    elsif @temp.is_a? Numeric
+      (@temp - 32) * 5.to_f / 9
+    else
+      @temp
```

Comment:
1) What will you need to modify to add Kelvin conversion? 
2) Do you see a problem with what you need to do for 1) ?
3) Can you find another solution that eases extensiblity (see SOLID principles from Trello)